### PR TITLE
feat: stabilize dark mode and add new dark palette 3/3

### PR DESCRIPTION
**Pull request by panvourtsis**
## Description

depends on https://github.com/Orfium/orfium-ictinus/pull/480

*This PR will be a part of 3 that has a goal for a small fix that I made for dark mode first try. 3/3*

This PR introduces a new dark mode config based on Figma and extends the current functions to apply a basic dark mode functionality to the app.

### TextField
Screen Recording 2022-03-30 at 10 59 06 AM

### Button
Screen Recording 2022-03-30 at 10 58 44 AM

### Checkbox
Screen Recording 2022-03-30 at 10 58 26 AM

## Test Plan

* Update state test because I changed the `darkBlue` to `blue`
* introduce `ThemeProvider.test.tsx` to test the dark mode switching

### CODE REVIEW

**Review by elektracodes — APPROVED** [review]

**Review by mkarajohn — CHANGES_REQUESTED** [review]
Please rebase on 480

**Comment by mkarajohn** [comment]
Without yet checking the code, shouldn't the icons turn black as well when dark mode is enabled? White icons on this background is not easily readable
Screen Recording 2022-03-30 at 10 58 44 AM

File: src/components/Radio/Radio.style.ts [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
 import { Props } from './Radio';
 
-const hoverColor = 'rgba(0, 0, 0, 0.05)';
+const lightHoverColor = 'rgba(14, 14, 23, 0.07)';
+const darkHoverColor = 'rgba(255, 255, 255, 0.1)';
 
-const focusedRadio = css`
-  box-shadow: 0 0 0 ${rem('12px')} ${hoverColor};
+const boxShadow = ({
+  inset = false,
```
Comment (SVonKarajann) [inline src/components/Radio/Radio.style.ts:13]: `inset` param in never used

File: src/components/Radio/Radio.style.ts [hunk 2]
```diff
@@ -80,8 +86,33 @@ export const customRadioStyles = (props: Pick<Props, 'checked' | 'disabled' | 'f
     width: 100%;
     height: 100%;
     box-sizing: border-box;
-    box-shadow: ${determineBoxShadow(props)(theme)};
     position: absolute;
+    // background-color: ${determineColorBasedOnState(props)(theme)};
```
Comment (SVonKarajann) [inline src/components/Radio/Radio.style.ts:90]: Remove if not necessary?

File: src/components/TextInputBase/config.ts [hunk 3]
```diff
@@ -11,42 +11,70 @@ export const MIN_WIDTH = 150;
...
+        border: {
+          width: 1,
+          color: {
+            default: {
+              name: 'darkGrey',
+              shade: 500,
+            },
+            pressed: {
+              name: 'blue',
+              shade: 350,
```
Comment (SVonKarajann) [inline src/components/TextInputBase/config.ts:69]: Is there any rule given from the DT on how we change shade values from light to dark mode?

**Review by mkarajohn — COMMENTED** [review]
Not approving because of https://github.com/Orfium/orfium-ictinus/pull/481#issuecomment-1084456291. Otherwise seems ok

**Comment by panvourtsis** [comment]
> Without yet checking the code, shouldn't the icons turn black as well when dark mode is enabled? White icons on this background are not easily readable Screen Recording 2022-03-30 at 10 58 44 AM

For this, there is no use of icons inside the component. There is a plan to replace this in the next versions. Until then the icons are passed as new components to the button so will be wrong to overwrite styles

File: src/components/TextInputBase/config.ts (hunk 3 above)
Comment (panvourtsis) [reply src/components/TextInputBase/config.ts:69]: whatever they have on the Figma rules on 4.5. This is not a final dark mode functionality but a very basic demo of how it works based on the DT specifications. Check the Figma for these

**Review by mkarajohn — APPROVED** [review]

**Pull request merged**